storage/src/store/base.rs

```diff
@@ -0,0 +1,845 @@
+//! A simple append-only key-value [`Store`], based on a log of state-change operations backed by a [`Journal`].
+
+use crate::{
+    index::Index,
+    journal::fixed::{Config as JConfig, Journal},
+    store::{operation::Fixed, Error, UpdateResult},
+    translator::Translator,
+};
+use commonware_runtime::{buffer::PoolRef, Clock, Metrics, Storage as RStorage};
+use commonware_utils::Array;
+use futures::{pin_mut, StreamExt};
+use tracing::{debug, warn};
+
+/// The size of the read buffer to use for replaying the operations log when rebuilding the
+/// snapshot.
+const SNAPSHOT_READ_BUFFER_SIZE: usize = 1 << 16;
+
+/// Configuration for initializing a [`Store`] database.
+#[derive(Clone)]
+pub struct Config<T: Translator> {
+    /// The name of the [`RStorage`] partition used to persist the (pruned) log of operations.
+    pub log_journal_partition: String,
+
+    /// The items per blob configuration value used by the log journal.
+    pub log_items_per_blob: u64,
+
+    /// The size of the write buffer to use for each blob in the log journal.
+    pub log_write_buffer: usize,
+
+    /// The [`Translator`] used by the compressed index.
+    pub translator: T,
+
+    /// The buffer pool to use for caching data.
+    pub buffer_pool: PoolRef,
+}
+
+/// A simple append-only key-value store backed by a log of operations.
+///
+/// The [`Store`] provides persistent key-value storage with transactional semantics and automatic
+/// recovery. Unlike [`crate::adb::any`], this store does not maintain an MMR (Merkle Mountain Range)
+/// for cryptographic authentication, making it significantly more efficient for local storage
+/// scenarios where proof generation is not required.
+///
+/// # Lifecycle
+///
+/// 1. **Initialization**: Create with [`Store::init`] using a [Config]
+/// 2. **Operations**: Use [`Store::update`] and [`Store::delete`] to modify state
+/// 3. **Queries**: Use [`Store::get`] to retrieve current values
+/// 4. **Persistence**: Call [`Store::commit`] to make changes durable
+/// 5. **Cleanup**: Call [`Store::close`] to shutdown gracefully or [`Store::destroy`] to remove all data
+///
+/// # Example
+///
+/// ```rust,ignore
+/// let mut store = Store::init(context, config).await?;
+///
+/// // Perform operations (not yet committed)
+/// store.update(key1, value1).await?;
+/// store.update(key2, value2).await?;
+/// store.delete(key1).await?;
+///
+/// // Query current state
+/// assert_eq!(store.get(&key1).await?, None);
+/// assert_eq!(store.get(&key2).await?, Some(value2));
+///
+/// // Commit to make changes persistent
+/// store.commit().await?;
+/// ```
+pub struct Store<E, K, V, T>
+where
+    E: RStorage + Clock + Metrics,
+    K: Array,
+    V: Array,
+    T: Translator,
+{
+    /// The [`Store`]'s log of operations.
+    ///
+    /// This log is expected to be pruned to the `pruned_to_loc` boundary and contain all subsequent operations.
+    pub(super) log: Journal<E, Fixed<K, V>>,
```

Comment:
I had originally intended for the `store` to not be integrated into `adb` (and use a `Variable<K, V>` here) but migrating `adb` now sounds tempting...